tests/std/tests/P2415R2_owning_view/test.cpp

```diff
@@ -0,0 +1,240 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <concepts>
+#include <memory>
+#include <ranges>
+#include <span>
+#include <utility>
+
+#include <range_algorithm_support.hpp>
+using namespace std;
+
+struct instantiator {
+    template <ranges::input_range R>
+    static constexpr void call() {
+        if constexpr (move_constructible<R>) {
+            using ranges::owning_view, ranges::begin, ranges::end, ranges::bidirectional_range, ranges::common_range,
+                ranges::contiguous_range, ranges::forward_range, ranges::input_range, ranges::random_access_range;
+            int input[3] = {0, 1, 2};
+
+            // properties
+            STATIC_ASSERT(ranges::enable_borrowed_range<owning_view<R>> == ranges::enable_borrowed_range<R>);
+
+            STATIC_ASSERT(ranges::view<owning_view<R>>);
+            STATIC_ASSERT(input_range<owning_view<R>>);
+            STATIC_ASSERT(forward_range<owning_view<R>> == forward_range<R>);
+            STATIC_ASSERT(bidirectional_range<owning_view<R>> == bidirectional_range<R>);
+            STATIC_ASSERT(random_access_range<owning_view<R>> == random_access_range<R>);
+            STATIC_ASSERT(contiguous_range<owning_view<R>> == contiguous_range<R>);
+
+            { // constructors and assignment operators
+                STATIC_ASSERT(constructible_from<owning_view<R>, R>);
+                STATIC_ASSERT(!constructible_from<owning_view<R>, R&>);
+                STATIC_ASSERT(!constructible_from<owning_view<R>, const R&>);
+                STATIC_ASSERT(!constructible_from<owning_view<R>, const R>);
+
+                STATIC_ASSERT(default_initializable<owning_view<R>> == default_initializable<R>);
+                STATIC_ASSERT(
+                    is_nothrow_default_constructible_v<owning_view<R>> == is_nothrow_default_constructible_v<R>);
+                if constexpr (default_initializable<R>) {
+                    [[maybe_unused]] owning_view<R> default_constructed;
+                }
+
+                owning_view<R> value_constructed{R{input}};

```

Comment:
Naming nitpick: After a variable named `default_constructed`, this name `value_constructed` seems confusing, given the Standard terminology of value-initialized.